### PR TITLE
GCE robotest

### PR DESCRIPTION
**Pull request by a-palchikov**
This PR adds support for running robotests on GCE.

The idea is to split the tests in two groups: regular PR test that should complete fast(er) and a nightly run that validates the master branch against a more detailed configuration.
This PR also adds initial support for nightly tests.

Complementary PR: https://github.com/gravitational/robotest/pull/134

### CODE REVIEW

File: lib/status/status.go [hunk 1]
```diff
@@ -105,22 +105,6 @@ func FromCluster(ctx context.Context, operator ops.Operator, cluster ops.Site, o
 	return status, nil
 }
 
-// Check classifies this cluster status as error
-func (s Status) Check() error {
```
Comment (a-palchikov) [inline lib/status/status.go]: Not sure why this was added. This breaks robotest's ability to quickly determine if an operation has failed.
Although `robotest` can use the full `gravity status` to do the same, the behavior we had with `--operation-id= --quiet` was to output just the operation status which does not work if the cluster is degraded and `Cluster status: degraded` is output instead.

File: tool/gravity/cli/app.go [hunk 2]
```diff
@@ -124,20 +124,31 @@ func listApps(env *localenv.LocalEnvironment, repository, appType string, showHi
...
+func uninstallApp(env *localenv.LocalEnvironment, appPackage *loc.Locator) error {
+	apps, err := env.SiteApps()
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	apps, err := env.AppServiceLocal(localenv.AppConfig{RegistryURL: registryURL})
-	if err != nil {
-		return trace.Wrap(err)
+
+	if appPackage == nil {
```
Comment (a-palchikov) [inline tool/gravity/cli/app.go]: Simplify 'app uninstall' to uninstall the cluster application if no package has been specified.

File: Jenkinsfile [hunk 3]
```diff
@@ -26,18 +26,18 @@ properties([
...
     string(name: 'PARALLEL_TESTS',
-           defaultValue: '80',
+           defaultValue: '4',
            description: 'Number of parallel tests to run.'),
     string(name: 'REPEAT_TESTS',
            defaultValue: '1',
            description: 'How many times to repeat each test.'),
     string(name: 'ROBOTEST_VERSION',
-           defaultValue: 'stable',
+           defaultValue: 'dima',
```
Comment (r0mant) [inline Jenkinsfile]: Just a reminder so you don't forget to change this once robotest gets merged.

File: assets/telekube/resources/app.yaml [hunk 4]
```diff
@@ -77,7 +77,7 @@ nodeProfiles:
         - name: rhel
           versions: ["7"]
         - name: ubuntu
-          versions: ["16.04"]
+          versions: ["16", "17"]
```
Comment (r0mant) [inline assets/telekube/resources/app.yaml]: Do we officially support Ubuntu 17? It's not even LTS Ubuntu release.

File: build.assets/robotest_run_nightly.sh [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+# gravity version -> list of OS releases to exercise on
+UPGRADE_MAP[5.0.24]="redhat:7 centos:7 debian:9 suse:12 ubuntu:16"
+
+readonly GET_GRAVITATIONAL_IO_APIKEY=${GET_GRAVITATIONAL_IO_APIKEY:?API key for distribution Ops Center required}
+readonly GRAVITY_BUILDDIR=${GRAVITY_BUILDDIR:?Set GRAVITY_BUILDDIR to the build directory}
+readonly ROBOTEST_SCRIPT=$(mktemp -d)/runsuite.sh
+
+# number of environment variables are expected to be set
+# see https://github.com/gravitational/robotest/blob/master/suite/README.md
+export ROBOTEST_VERSION=${ROBOTEST_VERSION:-dima}
```
Comment (r0mant) [inline build.assets/robotest_run_nightly.sh]: dima -> stable (once robotest is merged)

File: tool/gravity/cli/run.go [hunk 6]
```diff
@@ -438,9 +439,12 @@ func Execute(g *Application, cmd string, extraArgs []string) error {
 			*g.AppStatusCmd.Locator,
 			*g.AppStatusCmd.OpsCenterURL)
 	case g.AppUninstallCmd.FullCommand():
-		return uninstallApp(localEnv,
-			*g.AppUninstallCmd.Locator,
-			*g.AppUninstallCmd.OpsCenterURL)
+		var appPackage *loc.Locator
+		appPackage, err = loc.ParseLocator(*g.AppUninstallCmd.Locator)
```
Comment (r0mant) [inline tool/gravity/cli/run.go]: Won't this return an error right away if the locator isn't provided to the command? Which would render your changes for determining local app automatically above not having any effect?

**Comment by a-palchikov** [comment]
@knisbet About the nightlies. I've not focused on this yet, but I'm thinking of using Slack notifications and a dedicated channel for build failures feedback.

**Pull request merged**